### ISSUE TITLE
Unable to install package pyasic==0.40.3

### ISSUE THREAD

**Issue by oscarrdg** [Schnitzel/hass-miner#250]
Integration: 1.0.1
Home Assistant: 23.10.5

When trying to add a new entity I get an error:

Invalid config
The following integrations and platforms could not be set up:

miner (Show logs)
Please check your config and logs.

---
Logs:

Logger: homeassistant.util.package
Source: util/package.py:108
First occurred: 3:40:58 PM (3 occurrences)
Last logged: 3:41:35 PM

Unable to install package pyasic==0.40.3: ERROR: Cannot install pyasic==0.40.3 because these package versions have conflicting dependencies. ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts [notice] A new release of pip is available: 23.2.1 -> 23.3.1 [notice] To update, run: pip install --upgrade pip

---
I previously had installed the beta version 1.0.0-beta.
I don't know if it could be happening some conflict.

**Comment by UpstreamData** [comment]
Seems like it's conflicting with the environment with your old install of HASS-miner, are you able to uninstall it and re-install it to see if that fixes the issue?

**Comment by Schnitzel** [comment]
@oscarrdg you are not running the newest version of HomeAssisstant, can you upgrade to at least 2023.11.2 and try again?

**Comment by oscarrdg** [comment]
Uninstalled the integration, reinstalled again, but still getting the error:

Logger: homeassistant.util.package
Source: util/package.py:108
First occurred: 10:39:21 AM (3 occurrences)
Last logged: 10:39:56 AM

Unable to install package pyasic==0.40.3: ERROR: Cannot install pyasic==0.40.3 because these package versions have conflicting dependencies. ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts [notice] A new release of pip is available: 23.2.1 -> 23.3.1 [notice] To update, run: pip install --upgrade pip

---

I will update the Home Assistant to the last version and check again.

Thank you for your work!

**Comment by UpstreamData** [comment]
Now that I'm back home and am actually able to test, HACS seems to be really mad at the old versions for some reason.  I was able to fix this by removing HACS and re-installing.  Trouble is, HACS wouldn't let me remove it via the HASS interface, so I had to do it via ssh.  Here's the steps I used to set this back up -

**NOTE: THIS WILL KILL ALL HACS BASED INTEGRATIONS, DON'T DO THIS IF YOUR'RE WORRIED ABOUT THOSE**
- connect to the SSH/Terminal add on
- `cd homeassistant/custom_components` - Move to the custom components directory
- `rm -r miner` - Remove miner integration
- `rm -r hacs` - Remove HACS
- restart home assistant (HACS recommends you do this 2 times)
- re-install HACS the way you normally would, with the installation script
- remove the miner integration via HASS web UI if it still shows up in HACS after doing the install steps
- re-add the miner integration

Let me know if this works for you, I was running an even older version of the miner integration and this seemed to resolve it.
